cli/azd/pkg/auth/remote_credential.go

```diff
@@ -83,7 +83,7 @@ func (rc *RemoteCredential) GetToken(ctx context.Context, options policy.TokenRe
 	}
 
 	if err := json.NewDecoder(res.Body).Decode(&tokenResp); err != nil {
-		return azcore.AccessToken{}, fmt.Errorf("unmarshalling response: %w", err)
+		return azcore.AccessToken{}, remoteCredentialError(fmt.Errorf("unexpected format for token response: %w", err))
```

Comment:
nit: It's possible that the response was completely malformed (not just a format/contract mismatch). For example, "unexpected end of JSON". In this case, "decoding token response: unexpected end of JSON" may sound nicer than "unexpected format for token response: unexpected end of JSON"?

```suggestion
		return azcore.AccessToken{}, remoteCredentialError(fmt.Errorf("decoding token response: %w", err))
```